test/search/fragmentation_scheme_test.exs

```diff
@@ -0,0 +1,83 @@
+defmodule Search.FragmentationSchemeTest do
+  alias Search.FragmentationScheme
+  use ExUnit.Case, async: true
+
+  describe "split/2" do
+    test "when given an empty string, returns empty list" do
+      assert FragmentationScheme.split("") == []
+    end
+
+    test "when given a string which satisfies the size constraint, returns a singleton list with that string" do
+      str = "short string"
+
+      assert FragmentationScheme.split(str, max_size: 100) == [str]
+    end
+
+    test "when given a string that is too long and splitting along whitespace is possible, splits the string" do
+      str = "some words and some more words"
+
+      assert FragmentationScheme.split(str, max_size: 15) == [
+               "some words and ",
+               "some more words"
+             ]
+    end
+
+    test "when splitting along whitespace, respects non-space whitespace characters" do
+      str = "word\nword\tword\u{2003}word"
+
+      assert FragmentationScheme.split(str, max_size: 7) == [
+               "word\n",
+               "word\t",
+               "word\u{2003}",
+               "word"
+             ]
+    end
+
+    test "when splitting along whitespace and the text starts with whitespace, the whitespace characters are prepended to the first fragment" do
+      str = "    words and some more words"
+
+      assert FragmentationScheme.split(str, max_size: 15) == [
+               "    words and ",
+               "some more words"
+             ]
+    end
+
+    test "when cannot split along whitespace, splits along grapheme boundaries" do
+      str1 = "asdfghjkl"
+
+      assert FragmentationScheme.split(str1, max_size: 5) == [
```

Comment:
Out of curiosity, what max_size do you expect to use in production? Is max_size going to vary? If not, perhaps we don't need the `--max-size` CLI option?

Reply:
For the small models used here it would be the context size of the models so that information from the fragments isn't discarded when trimming the token tensor to the context size of the embedding model, but in production the context size would probably be very large, so it would be more of a hyperparameter. From my previous experiments, the results from fragmentation can influence lookup precision in unexpected ways, so it will probably have to be benchmarked on a per-model basis. In general, tuning the max_size value provides control over the balance of how specific the search results are and how precise they are